### PR TITLE
Adding File-dialog

### PR DESCRIPTION
**Pull request by JijaLoqie**

### CODE REVIEW

File: src/Utilities/FileSystem/FileManager.cpp [hunk 1]
```diff
@@ -37,7 +37,20 @@
 namespace MxEngine
 {
 
+    MxString FileManager::OpenFileDialog()
```
Comment (MomoDeve) [inline src/Utilities/FileSystem/FileManager.cpp]: add the ability to pass filter string as parameter to OpenFileDialog

File: src/Utilities/FileSystem/FileManager.cpp [hunk 2]
```diff
@@ -37,7 +37,20 @@
...
+    {
+        auto selection = pfd::open_file("Select a file", ".",
+            { "Image Files", "*.png *.jpg *.jpeg *.bmp *.tga *.hdr",
+              "All Files", "*" },
+            pfd::opt::multiselect).result();
+        return selection.empty() ? "" : ToMxString(selection.front());
 
+    }
+    MxString FileManager::OpenFileDialogForMesh()
```
Comment (MomoDeve) [inline src/Utilities/FileSystem/FileManager.cpp]: this should not be a separate function

File: src/Utilities/FileSystem/FileManager.cpp (hunk 1 above)
Comment (MomoDeve) [reply src/Utilities/FileSystem/FileManager.cpp]: add the ability to specify start directory. not just `.`

File: src/Utilities/ImGui/Editors/ResourceEditor.cpp [hunk 3]
```diff
@@ -48,13 +51,19 @@ namespace MxEngine::GUI
         if (ImGui::CollapsingHeader("create new texture"))
         {
             static MxString path;
-            if (GUI::InputTextOnClick("load from path", path, 128))
+            if (ImGui::Button("load from file"))
             {
-                auto newTexture = GraphicFactory::Create<Texture>();
-                newTexture->Load(path);
-                newTexture.MakeStatic();
+                path = FileManager::OpenFileDialog();
```
Comment (MomoDeve) [inline src/Utilities/ImGui/Editors/ResourceEditor.cpp]: make `path` local variable (see all cases)

File: CMakeLists.txt [hunk 4]
```diff
@@ -38,4 +38,5 @@ if (MXENGINE_BUILD_SAMPLES)
     add_subdirectory(samples/VRCameraSample)
     add_subdirectory(samples/GrassSample)
     add_subdirectory(samples/FluidSimulation)
+    add_subdirectory(samples/Dimon)
```
Comment (MomoDeve) [inline CMakeLists.txt]: delete this sample from pull request

**Review by MomoDeve — CHANGES_REQUESTED** [review]

File: src/Utilities/FileSystem/FileManager.cpp [hunk 5]
```diff
@@ -37,20 +37,21 @@
...
+        std::vector<std::string> selection;
+        if (ForMesh) {
+            selection = pfd::open_file("Select a file").result();
+        }
+        else {
+            selection = pfd::open_file("Select a file", ".",
+                { "Image Files", "*.png *.jpg *.jpeg *.bmp *.tga *.hdr",
+                  "All Files", "*" },
+                pfd::opt::multiselect).result();
+        }
```
Comment (WhiteBlackGoose) [inline src/Utilities/FileSystem/FileManager.cpp]: Wouldn't it be a better idea to pass this selection as an argument to the function?

File: src/Utilities/FileSystem/FileManager.cpp [hunk 6]
```diff
@@ -37,20 +37,21 @@
...
         return selection.empty() ? "" : ToMxString(selection.front());
 
     }
-    MxString FileManager::OpenFileDialogForMesh()
```
Comment (WhiteBlackGoose) [inline src/Utilities/FileSystem/FileManager.cpp]: I feel like this function was OK, it just should call `OpenFileDialog` with an appropriate selection

File: src/Utilities/ImGui/Editors/ComponentEditors/RenderingEditors.cpp [hunk 7]
```diff
@@ -73,7 +73,7 @@ namespace MxEngine::GUI
 		static MxString path;
 		if (ImGui::Button("load from file"))
 		{
-			path = FileManager::OpenFileDialog();
+			path = FileManager::OpenFileDialog(1);
```
Comment (WhiteBlackGoose) [inline src/Utilities/ImGui/Editors/ComponentEditors/RenderingEditors.cpp]: 1 for bool -____-

**Review by WhiteBlackGoose — COMMENTED** [review]
I'm not a cpper, but I have a few feelings...

File: src/Utilities/ImGui/Editors/ResourceEditor.cpp [hunk 8]
```diff
@@ -197,10 +195,9 @@ namespace MxEngine::GUI
             ImGui::Text("empty resource");
         }
         
-        static MxString path;
         if (ImGui::Button("load from file"))
         {
-            path = FileManager::OpenFileDialog(0);
+            MxString path = FileManager::OpenFileDialog("*.png *.jpg *.jpeg *.bmp *.tga *.hdr");
```
Comment (WhiteBlackGoose) [inline src/Utilities/ImGui/Editors/ResourceEditor.cpp]: If you ask me, I'd either abstract it in another function like `OpenFileDialogImage` or make a constant, especially if you use this string multiple times

File: src/Utilities/FileSystem/FileManager.h [hunk 9]
```diff
@@ -47,7 +47,7 @@ namespace MxEngine
         inline static FileManagerImpl* manager = nullptr;
         static void AddDirectory(const FilePath& directory);
     public:
-        static MxString OpenFileDialog(bool ForMesh);
+        static MxString OpenFileDialog(const std::string& types);
```
Comment (WhiteBlackGoose) [inline src/Utilities/FileSystem/FileManager.h]: Shouldn't we stick to `MxString`?

**Review by WhiteBlackGoose — COMMENTED** [review]
Nice, I like it more

**Pull request merged**